src/m365/teams/commands/meeting/meeting-transcript-list.ts

```diff
@@ -0,0 +1,140 @@
+import { Cli } from '../../../../cli/Cli';
+import Command from '../../../../Command';
+import { Logger } from '../../../../cli/Logger';
+import GlobalOptions from '../../../../GlobalOptions';
+import GraphCommand from "../../../base/GraphCommand";
+import commands from '../../commands';
+import { odata } from '../../../../utils/odata';
+import { validation } from '../../../../utils/validation';
+import * as AadUserGetCommand from '../../../aad/commands/user/user-get';
+import { Options as AadUserGetCommandOptions } from '../../../aad/commands/user/user-get';
+import { accessToken } from '../../../../utils/accessToken';
+
+interface CommandArgs {
+  options: Options;
+}
+
+interface Options extends GlobalOptions {
+  userId?: string;
+  userName?: string;
+  email?: string;
+  meetingId: string;
+}
+
+class TeamsMeetingTranscriptListCommand extends GraphCommand {
+  public get name(): string {
+    return commands.MEETING_TRANSCRIPT_LIST;
+  }
+
+  public get description(): string {
+    return 'Lists all transcripts for a given meeting';
+  }
+
+  public defaultProperties(): string[] | undefined {
+    return ['id', 'createdDateTime'];
+  }
+
+  constructor() {
+    super();
+
+    this.#initTelemetry();
+    this.#initOptions();
+    this.#initValidators();
+  }
+
+  #initTelemetry(): void {
+    this.telemetry.push((args: CommandArgs) => {
+      Object.assign(this.telemetryProperties, {
+        userId: typeof args.options.userId !== 'undefined',
+        userName: typeof args.options.userName !== 'undefined',
+        email: typeof args.options.email !== 'undefined'
+      });
+    });
+  }
+
+  #initOptions(): void {
+    this.options.unshift(
+      {
+        option: '-u, --userId [userId]'
+      },
+      {
+        option: '-n, --userName [userName]'
+      },
+      {
+        option: '--email [email]'
+      },
+      {
+        option: '-m, --meetingId <meetingId>'
+      }
+    );
+  }
+
+  #initValidators(): void {
+    this.validators.push(
+      async (args: CommandArgs) => {
+        if (args.options.userId && !validation.isValidGuid(args.options.userId)) {
+          return `${args.options.userId} is not a valid Guid`;
+        }
+
+        return true;
+      }
+    );
+  }
+
+  public async commandAction(logger: Logger, args: CommandArgs): Promise<void> {
+    const isAppOnlyAuth: boolean = accessToken.isAppOnlyAccessToken(auth.service.accessTokens[this.resource].accessToken);
+    if (isAppOnlyAuth && !args.options.userId && !args.options.userName && !args.options.email) {
+      this.handleError(`The option 'userId', 'userName' or 'email' is required when retrieving meeting transcript list using app only permissions`);
+    }
+    else if (!isAppOnlyAuth && (args.options.userId || args.options.userName || args.options.email)) {
+      this.handleError(`The options 'userId', 'userName' and 'email' cannot be used when retrieving meeting transcript list using delegated permissions`);
+    }
```

Comment:
Let us remove this check.

An app could have both delegated and application permissions. A person should decide what they want to do based on their permissions